jupyter_server/extension/application.py

```diff
@@ -256,15 +288,21 @@ def initialize(self, serverapp, argv=[]):
         self._prepare_settings()
         self._prepare_handlers()
 
-    def start(self, **kwargs):
+    def start(self):
         """Start the underlying Jupyter server.
         
         Server should be started after extension is initialized.
         """
-        # Start the browser at this extensions default_url.
-        self.serverapp.default_url = self.default_url
+        # Override the browser open file to 
+        # Override the server's display url to show extension's display URL.
+        self.serverapp.custom_display_url = self.custom_display_url
+        # Override the server's default option and open a broswer window.
+        self.serverapp.open_browser = True
+        # Hijack the server's browser-open file to land on
+        # the extensions home page.
+        self.serverapp._write_browser_open_file = self._write_browser_open_file
```

Comment:
Hm, maybe. Let me explain what my reason:

I found two ways for an extension to reroute the server landing page.
1. Change the server's `default_url` and have it land on the extension. There were two challenges with this approach:
    - The server's `default_url` needs to be set to `<extension>.default_url` before the `server.initialize()` is called, otherwise we have to change a bunch of downstream trails of `default_url` in serverapp. The issue is that `<extension>.default_url` doesn't get set until after server is initialized. 
    - The server's `default_url` truly is a different trait than the extension's `default_url`. In my mind, they could/should point to two different urls (though, I don't have an example of when someone might want to do this right now).
2. Change the server's browser open file. I chose this approach because:
    - The server doesn't open a browser window by default, anyways. In fact, a browser window doesn't really make sense without an `ExtensionApp`.
    - This file is only created if the server was started by an extension. In that case, I think it makes sense that the server's browser file *always* opens to the extension's homepage.
    - The `ServerApp.default_url` and `ExtensionApp.default_url` can be pointing to different URLs (maybe a plus?).

Reply:
When I wrote "wrong way', I meant to overwrite a protected method on the app. This is basically monkey-patching, and it means that there isn't really a clear API boundary between the two objects.